arangod/Pregel/PregelFeature.cpp

```diff
@@ -650,89 +658,146 @@ auto PregelFeature::cleanupWorker(ExecutionNumber executionNumber)
   });
 }
 
-void PregelFeature::handleConductorRequest(TRI_vocbase_t& vocbase,
-                                           std::string const& path,
-                                           VPackSlice const& body,
-                                           VPackBuilder& outBuilder) {
+auto PregelFeature::process(ModernMessage message, TRI_vocbase_t& vocbase)
+    -> ResultT<ModernMessage> {
   if (isStopping()) {
-    return;  // shutdown ongoing
+    return Result{TRI_ERROR_INTERNAL, "Pregel is stopping"};
   }
+  auto response = apply(message.executionNumber, message.payload, vocbase);
+  if (response.fail()) {
+    VPackBuilder serialized;
+    serialize(serialized, message);
+    return Result{TRI_ERROR_CURSOR_NOT_FOUND,
+                  fmt::format("Processing request failed: Execution {}: {}: {}",
+                              message.executionNumber, response.errorMessage(),
+                              serialized.toJson())};
+  }
+  return ModernMessage{.executionNumber = message.executionNumber,
+                       .payload = response.get()};
+}
 
-  if (path != Utils::modernMessagingPath) {
-    THROW_ARANGO_EXCEPTION_MESSAGE(
-        TRI_ERROR_CURSOR_NOT_FOUND,
-        fmt::format("Conductor path not found: {}", path));
-  }
+auto PregelFeature::apply(ExecutionNumber const& executionNumber,
+                          MessagePayload const& message, TRI_vocbase_t& vocbase)
+    -> ResultT<MessagePayload> {
+  Scheduler* scheduler = SchedulerFeature::SCHEDULER;
+  return std::visit(
+      overloaded{
+          [&](CreateWorker const& x) -> ResultT<MessagePayload> {
+            try {
+              auto created = AlgoRegistry::createWorker(vocbase, x, *this);
+              TRI_ASSERT(created.get() != nullptr);
+              addWorker(std::move(created), executionNumber);
+              return {
+                  WorkerCreated{.senderId = ServerState::instance()->getId()}};
+            } catch (basics::Exception& e) {
+              return Result{e.code(), e.message()};
+            }
+          },
+          [&](LoadGraph const& x) -> ResultT<MessagePayload> {
+            auto w = worker(executionNumber);
```

Comment:
There's a lot of duplication going on here, but I could not yet think of a way that solves this problem just now;

As a longer-term solution the destination of the message should contain the execution number, or better yet, we just know which workers we started as conductor and speak to them directly.